File: core/tmp/Python/bit-operation/counting_bits.py
from typing import List

def countBits(num: int) -> List[int]:
    remain = [0 for _ in range(num + 1)]
    for i in range(1, num + 1):
        temp = i & (i - 1)
        remain[i] = remain[temp] + 1
    
    return remain

File: core/tmp/Java/src/com/dbc/algorithm/BitOperation/CountingBits.java
package com.dbc.algorithm.BitOperation;

public class CountingBits {
    public int[] countBits(int num) {
        int[] remain = new int[num + 1];
        for (int i = 1; i < num + 1; i++) {
            int temp = i & (i - 1);
            remain[i] = 1 + remain[temp];
        }
        return remain;
    }
}
